<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>770</width>
    <height>675</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>770</width>
    <height>675</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>770</width>
    <height>675</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>3D Viewer v1.0</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>60</y>
      <width>160</width>
      <height>3</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>10</y>
      <width>160</width>
      <height>3</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>591</x>
      <y>10</y>
      <width>16</width>
      <height>170</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>10</y>
      <width>23</width>
      <height>170</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="MyOpenGLWidget" name="oglwidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>580</width>
      <height>362</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="load_obj_button">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>20</y>
      <width>121</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Load .obj file..</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>71</y>
      <width>83</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Имя фаила:</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>117</y>
      <width>61</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Вершин:</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>140</y>
      <width>48</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Ребер:</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="name_label">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>94</y>
      <width>110</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="vertices_label">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>116</y>
      <width>60</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="faces_label">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>140</y>
      <width>60</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="save_in_jpeg">
    <property name="geometry">
     <rect>
      <x>598</x>
      <y>340</y>
      <width>162</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сохранить в jpeg</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_in_bmp">
    <property name="geometry">
     <rect>
      <x>598</x>
      <y>260</y>
      <width>162</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сохранить в bmp</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_in_gif">
    <property name="geometry">
     <rect>
      <x>598</x>
      <y>300</y>
      <width>162</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сохранить в gif</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>378</y>
      <width>360</width>
      <height>230</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="3" column="1">
      <widget class="QLabel" name="scale">
       <property name="text">
        <string>1.00</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="6" column="1" colspan="2">
      <widget class="QSlider" name="Z_trans_slider">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>-10000</number>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Передвижение по z:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Передвижение по х:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Передвижение по у:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Вращение по оси х:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="z_angle">
       <property name="text">
        <string>0°</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QSlider" name="Y_rotation_slider">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="4" column="1" colspan="2">
      <widget class="QSlider" name="X_trans_slider">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>-10000</number>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QSlider" name="scaling_slider">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>100</number>
       </property>
       <property name="maximum">
        <number>3000</number>
       </property>
       <property name="singleStep">
        <number>50</number>
       </property>
       <property name="value">
        <number>1000</number>
       </property>
       <property name="tracking">
        <bool>true</bool>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Масштабирование:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="x_angle">
       <property name="layoutDirection">
        <enum>Qt::LeftToRight</enum>
       </property>
       <property name="text">
        <string>0°</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Вращение по оси z:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Вращениепо оси у:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="y_angle">
       <property name="text">
        <string>0°</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="5" column="1" colspan="2">
      <widget class="QSlider" name="Y_trans_slider">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>-10000</number>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QSlider" name="X_rotation_slider">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QSlider" name="Z_rotation_slider">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>382</x>
      <y>515</y>
      <width>100</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвета вершин:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="reset_position">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>610</y>
      <width>360</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сбросить  аффинные преобразования</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>383</x>
      <y>549</y>
      <width>100</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвет фона:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>382</x>
      <y>582</y>
      <width>100</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Цвет линий:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_settings">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>610</y>
      <width>190</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Загрузить настройки</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_settings">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>610</y>
      <width>179</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Сохранить настройки</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>381</x>
      <y>408</y>
      <width>377</width>
      <height>34</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_13">
       <property name="text">
        <string>Отображение линий:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="not_stipple_line">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Сплошная</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="stipple_line">
       <property name="text">
        <string>Пунктирная</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>381</x>
      <y>378</y>
      <width>376</width>
      <height>30</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>Тип проекции:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="parallel_proj">
       <property name="text">
        <string>Параллельная</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="central_proj">
       <property name="text">
        <string>Центральная</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>599</x>
      <y>179</y>
      <width>161</width>
      <height>3</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>488</x>
      <y>511</y>
      <width>271</width>
      <height>34</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QSlider" name="vert_color_R">
       <property name="tabletTracking">
        <bool>false</bool>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="sliderPosition">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="vert_color_G">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="sliderPosition">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="vert_color_B">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="sliderPosition">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>489</x>
      <y>545</y>
      <width>271</width>
      <height>34</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QSlider" name="back_color_R">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="back_color_G">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="back_color_B">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>489</x>
      <y>579</y>
      <width>271</width>
      <height>30</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QSlider" name="line_color_R">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>10</number>
       </property>
       <property name="sliderPosition">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="line_color_G">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="sliderPosition">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="line_color_B">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="sliderPosition">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>476</y>
      <width>379</width>
      <height>35</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>Толщина линий:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="line_width_slider">
       <property name="maximum">
        <number>30</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_15">
       <property name="text">
        <string>Размер вершин:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="vertex_width_slider">
       <property name="maximum">
        <number>30</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>440</y>
      <width>379</width>
      <height>36</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QLabel" name="label_19">
       <property name="text">
        <string>Отображение вершин:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="circle_vert_radio">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Круг</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="square_vert_radio">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Квадрат</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="disable_vert_radio">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Нет</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>770</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyOpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myopenglwidget.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>X_rotation_slider</tabstop>
  <tabstop>Y_rotation_slider</tabstop>
  <tabstop>Z_rotation_slider</tabstop>
  <tabstop>scaling_slider</tabstop>
  <tabstop>X_trans_slider</tabstop>
  <tabstop>Y_trans_slider</tabstop>
  <tabstop>Z_trans_slider</tabstop>
  <tabstop>load_obj_button</tabstop>
  <tabstop>back_color_G</tabstop>
  <tabstop>back_color_B</tabstop>
  <tabstop>save_settings</tabstop>
  <tabstop>vertex_width_slider</tabstop>
  <tabstop>line_color_R</tabstop>
  <tabstop>line_color_G</tabstop>
  <tabstop>line_color_B</tabstop>
  <tabstop>not_stipple_line</tabstop>
  <tabstop>stipple_line</tabstop>
  <tabstop>load_settings</tabstop>
  <tabstop>save_in_jpeg</tabstop>
  <tabstop>save_in_bmp</tabstop>
  <tabstop>save_in_gif</tabstop>
  <tabstop>parallel_proj</tabstop>
  <tabstop>central_proj</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
